### PR TITLE
[WINDOWS] Add "/bigobj" that solves compile issue during debug

### PR DESCRIPTION
**Pull request by maiadegraaf**
DuckDB was not compiling on Windows in debug mode and was throwing this [error](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1128?view=msvc-170). Adding the `/bigobj` flag fixed it.

### CODE REVIEW

**Comment by maiadegraaf** [comment]
Fixes:
https://github.com/duckdb/duckdb/issues/10683

File: CMakeLists.txt [hunk 1]
```diff
@@ -94,6 +94,12 @@ if(${FORCE_COLORED_OUTPUT})
   endif()
 endif()
 
+#Solves compile issue for debug builds
+#https://learn.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1128?view=msvc-170
+if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug" AND MSVC)
+  add_compile_options(/bigobj)
```
Comment (Mytherin) [inline CMakeLists.txt:100]: We have this definition for amalgamation builds in `src/CMakeLists.txt:36` - perhaps we can just move it up one layer? I think we might need it for more than just debug builds, and it doesn't hurt to enable for all Windows builds

**Comment by Mytherin** [comment]
Hm, looks like the MINGW builds are not happy now. Perhaps this should be `if (MSVC)` instead of `if (WIN32)`?

**Comment by maiadegraaf** [comment]
I'll run the CI on my branch first and see what happens.

**Pull request merged**